common/v2/features/Dashboard/components/AccountDropdown.tsx

```diff
@@ -60,22 +61,31 @@ const SDropdown = dropdown`
 
 const renderAccounts = (
   accounts: ExtendedAccount[],
+  labels: AddressBook[],
   selected: string[],
   handleChange: (uuid: string) => void
 ) =>
-  accounts.map(({ uuid, label, address }: ExtendedAccount) => (
-    <Checkbox
-      key={uuid}
-      name={`account-${uuid}`}
-      checked={selected.includes(uuid)}
-      onChange={() => handleChange(uuid)}
-      label={label}
-      icon={() => <Identicon className="AccountDropdown-menu-identicon" address={address} />}
-    />
-  ));
+  accounts.map(({ uuid, address }: ExtendedAccount) => {
+    const detectedLabel: AddressBook | undefined = labels.find(
```

Comment:
1. Naming is important (and hard). When you right 'detected' I'm thinking of listening, scanning, watching, so imagining that you are catching something that is in movement. This is just more like a 'join' between structures. ~~Can we just use `accountLabel` | `label` instead?~~

I see that you need two variable names. How about

```
const addressCard = labels.find(...)
const addressLabel = addressCard ? addressCard.label : 'Unknown Account'
```

2. I thought we had the 'label' included in the account?
`.find()` returns only the first match, and it's my understanding that we do the address+network combination because the same address can be used on different networks. If we store the label under addressBook, we will be unable to provide a different label for address+network1 AND address+network2. And if we don't support this usecase we can simplify a lot of our data structure

3. I see you have a helper function `getLabelByAcocunt` let's use it here as well!

Reply:
Re:
> I thought we had the 'label' included in the account?

That's what im trying to remove in this PR. I'm tying labels to a mixture of network & address so that we don't have to query for labels from 2 different places when doing lookups.